packages/medusa/src/services/cart.ts

```diff
@@ -570,10 +579,24 @@ class CartService extends BaseService {
         }
       }
 
-      const result = await this.retrieve(cartId)
+      const result = await this.retrieve(cartId, {
+        relations: ["items", "discounts", "discounts.rule"],
+      })
+
+      // delete all old line item adjustments
+      await this.lineItemAdjustmentService_
+        .withTransaction(manager)
+        .delete({ item_id: result.items.map((item) => item.id) })
+
+      // potentially create/update line item adjustments
+      await this.lineItemAdjustmentService_
+        .withTransaction(manager)
+        .createAdjustments(result)
```

Comment:
One area where we might have to double-check if we are good is the case where a swap is created and the swap cart has a return line. The existing adjustments on this line should not be deleted as they will have come with the discounts that were applied when the original order was placed.

We can maybe do `.delete({ item_id:. result.items.filter(i => !i.is_return).map(...) })`?

Reply:
well spotted! will fix, thanks!